theano/gof/fg.py

```diff
@@ -488,8 +497,9 @@ def find_path_to(output_var, input_var):
                     self.__setup_r__(input)
                     self.variables.add(input)
                 self.__add_clients__(input, [(node, i)])
-            assert node.fgraph is self
-            self.execute_callbacks('on_import', node, reason)
+            if not prevent_addition :
```

Comment:
I'm trying to prevent the once removed ```apply_node``` from getting added back to the graph. Should I raise an error in this case ?